ibm/resource_ibm_cdn.go

```diff
@@ -0,0 +1,579 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/softlayer/softlayer-go/sl"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/services"
+)
+
+const str string = ".cdnedge.bluemix.net"
+
+func resourceIBMCDN() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceIBMCDNCreate,
+		Read:   resourceIBMCDNRead,
+		Update: resourceIBMCDNUpdate,
+		Delete: resourceIBMCDNDelete,
+		Exists: resourceIBMCDNExists,
+
+		Schema: map[string]*schema.Schema{
+			"hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"vendor_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"origin_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "HOST_SERVER",
+			},
+			"origin_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"bucketname": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"protocol": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "HTTP",
+			},
+			"httpport": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  80,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"httpsport": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  443,
+			},
+			"cname": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  false,
+			},
+			"header": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"respectheaders": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"fileextension": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"certificatetype": &schema.Schema{
```

Comment:
https://cloud.ibm.com/docs/infrastructure/CDN?topic=CDN-cdn-api-reference#createdomainmapping based on this it can be of SHARED_SAN_CERT or WILDCARD_CERT. Remove Default "0" and add validateFunc